docs/configexamples/ansible.rst

```diff
@@ -0,0 +1,302 @@
+
+.. image:: /_static/images/ansible.png
+   :width: 80%
+   :align: center
+   :alt: Network Topology Diagram
+
+We have four pre-configured routers with this configuration:
+
+.. code-block:: none
+
+	set interfaces ethernet eth0 address dhcp
+	set service ssh
+	commit
+	save
+
+* vyos7 - 192.0.2.105
+* vyos8 - 192.0.2.106
+* vyos9 - 192.0.2.107
+* vyos10 - 192.0.2.108
+
+Install the Ansible:
+====================
+.. code-block:: none
+
+	# apt-get install ansible
+	Do you want to continue? [Y/n] y
+
+Install the paramiko:
+=====================
+
+.. code-block:: none
+
+	#apt-get install -y python3-paramiko
+
+Check the version:
+==================
+
+.. code-block:: none
+
+	# ansible --version
+	ansible 2.10.8
+	config file = None
+	configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
+	ansible python module location = /usr/lib/python3/dist-packages/ansible
+	executable location = /usr/bin/ansible
+	python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110]
+
+Basik configuration of the ansible.cfg:
+=======================================
+
+.. code-block:: none
+
+	# nano /root/ansible.cfg
+	[defaults]
+	host_key_checking = no
+
+Add all the hosts of VyOS:
+==========================
+
+.. code-block:: none
+
+	# nano /root/hosts
+	[vyos7_host]
+	vyos7 ansible_ssh_host=192.0.2.105
+	[vyos7_host:vars]
+	ansible_python_interpreter=/usr/bin/python3
+	ansible_user=vyos
+	ansible_ssh_pass=vyos
+	ansible_network_os=vyos
+	ansible_connection=network_cli
+	
+	[vyos8_host]
+	vyos8 ansible_ssh_host=192.0.2.106
+	[vyos8_host:vars]
+	ansible_python_interpreter=/usr/bin/python3
+	ansible_user=vyos
+	ansible_ssh_pass=vyos
+	ansible_network_os=vyos
+	ansible_connection=network_cli
+	
+	[vyos9_host]
+	vyos9 ansible_ssh_host=192.0.2.107
+	[vyos9_host:vars]
+	ansible_python_interpreter=/usr/bin/python3
+	ansible_user=vyos
+	ansible_ssh_pass=vyos
+	ansible_network_os=vyos
+	ansible_connection=network_cli
+	
+	[vyos10_host]
+	vyos10 ansible_ssh_host=192.0.2.108
+	[vyos10_host:vars]
+	ansible_python_interpreter=/usr/bin/python3
+	ansible_user=vyos
+	ansible_ssh_pass=vyos
+	ansible_network_os=vyos
+	ansible_connection=network_cli
+
+Add the simple playbook with the tasks for each router:
+=======================================================
+
+.. code-block:: none
+
+	# nano /root/main.yml
+	
+	---
+	- hosts: vyos7
+	
+	connection: network_cli
+	gather_facts: 'no'
+	
+	tasks:
+		- name: Configure remote 7
+		vyos_config:
+			lines:
+			- set system host-name vyos7
+			- set system name-server 8.8.8.8
+			- set interfaces ethernet eth0 description WAN
+			- set interfaces ethernet eth1 description LAN
+			- set interfaces ethernet eth2 disable
+			- set interfaces ethernet eth3 disable
+			save:
+			true
+	
+	- hosts: vyos8
+	
+	connection: network_cli
+	gather_facts: 'no'
+	
+	tasks:
+		- name: Configure remote 8
+		vyos_config:
+			lines:
+			- set system host-name vyos8
```

Comment:
Why not use “ {{ inventory_hostname }}” as you set it in inventory file?

Reply:
I'll try it and make changes